src/include/migraphx/op/reshape.hpp

```diff
@@ -64,45 +65,83 @@ struct reshape
 
     std::string name() const { return "reshape"; }
 
-    shape dyn_compute_shape(shape s0) const
+    // Assumes that the shape from the `dims` attribute will be valid at run-time.
+    // Makes no checks for the validity of the `dims` attribute for the given input shape.
+    shape dyn_1arg_compute_shape(shape s0) const
     {
-        auto dyn_dims = s0.dyn_dims();
-        // track number of fixed elements in input and output
-        std::size_t num_dims_ele = 1;
-        std::size_t num_dd_ele   = 1;
-        for(std::size_t i = 0; i < dyn_dims.size(); ++i)
+        auto input_dyn_dims = s0.dyn_dims();
+        bool has_negative_dim_attr = false;
+        // construct output dynamic shape from dims attribute
+        std::vector<shape::dynamic_dimension> output_dyn_dims(dims.size());
+        for(int i = 0; i < dims.size(); ++i)
         {
-            if(dyn_dims[i].is_fixed())
+            int64_t d = dims.at(i);
+            if(d == 0)
+            {
+                output_dyn_dims.at(i) = input_dyn_dims.at(i);
+            }
+            else if(d == -1)
```

Comment:
Can there be two indices of value -1 -- maybe checked for error handling? Thanks.

Reply:
No, there can only be one -1 dimension. We check that there is only one -1 dimension earlier during parsing.